aten/src/ATen/StorageImpl.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <ATen/Scalar.h>
+
+#include <ATen/Allocator.h>
+#include <ATen/ScalarType.h>
+#include <ATen/ScalarTypeUtils.h>
+#include <TH/THTypeConversion.hpp>
+#include <atomic>
+
+// Note [Weak references for intrusive refcounting]
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// Here's the scheme:
+//
+//  - refcount == number of strong references to the object
+//    weakcount == number of weak references to the object,
+//      plus one more if refcount > 0
+//
+//  - THStorageImpl stays live as long as there are any strong
```

Comment:
is THStorageImpl a thing?

Reply:
No, sorry, I'll delete all this garbage.

Reply:
Don't delete the entire note, it's still valid